src/cli/diagnostics.py

```
@@ -229,7 +309,479 @@ def run_workspace(root: Path, *, write: bool, output: Optional[Path]) -> int:
+    Raises:
+        ProbeFailure: If repeated rate-limit errors occur for `max_attempts` attempts, with remediation guidance and aggregated error details.
+        Exception: Any non-rate-limit exception raised by `operation` is propagated unchanged.
+    """
+    attempt = 0
+    delay = base_delay
+    errors: List[str] = []
+    while attempt < max_attempts:
+        attempt += 1
+        try:
+            return operation()
+        except Exception as exc:  # noqa: BLE001 - controlled handling
```

Ruff (0.13.1)

Unused `noqa` directive (unused: `BLE001`)

Remove unused `noqa` directive

(RUF100)